src/main/java/pl/taskyers/taskybase/project/slo/ProjectInvitationSLOImpl.java

```diff
@@ -0,0 +1,85 @@
+    
+    private final TokenSLO projectInvitationTokenSLO;
+    
+    @Override
+    public ResponseEntity sendEmailWithInvitationToken(String userName, String projectName) {
+        if ( userSLO.getEntityByUsername(userName).isPresent() && projectSLO.getProjectEntityByName(projectName).isPresent() ) {
+            UserEntity userEntity = userSLO.getEntityByUsername(userName).get();
+            ProjectEntity projectEntity = projectSLO.getProjectEntityByName(projectName).get();
+            if ( roleSLO.hasPermission(userEntity, projectEntity, "project.invite.others") ) {
+                return ResponseEntity.status(HttpStatus.FORBIDDEN)
+                        .body(new ResponseMessage<String>(MessageCode.project_permission_not_granted.getMessage(), MessageType.WARN));
+            }
+            projectInvitationTokenSLO.createToken(userEntity, projectEntity);
+            boolean emailWasSent = emailSLO.sendEmailWithTemplateToSingleAddressee(AccountConverter.convertToDTO(userEntity),
+                    MessageCode.email_subject_project_invitation.getMessage(),
+                    EmailConstants.PROJECT_INVITATION_PATH,
+                    new String[]{ "name", "surname", "projectName", "token" },
+                    new Object[]{ userEntity.getName(), userEntity.getSurname(), projectEntity.getName(),
+                            EmailConstants.PROJECT_INVITATION_URL_TOKEN.replace("{tokenPlaceholder}",
+                                    projectInvitationTokenSLO.getToken(userEntity)) });
+            
+            return emailWasSent ?
+                    ResponseEntity.ok(
+                            new ResponseMessage<String>(MessageCode.email_with_token_sent.getMessage(userEntity.getEmail()), MessageType.SUCCESS)) :
+                    ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+                            .body(new ResponseMessage<>(MessageCode.server_problem_occured.getMessage(), MessageType.ERROR));
+        }
+        return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                .body(new ResponseMessage<>(MessageCode.field_not_found.getMessage("Username", userName), MessageType.WARN));
+    }
+    
+    @Override
+    public ResponseEntity acceptInvitation(String token) {
+        ProjectInvitationTokenEntity projectInvitationTokenEntity = (ProjectInvitationTokenEntity) projectInvitationTokenSLO.getTokenEntity(token);
+        if ( projectInvitationTokenEntity == null ) {
+            log.warn("Token " + token + " was not found");
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                    .body(new ResponseMessage<String>(MessageCode.field_not_found.getMessage("Token", token), MessageType.WARN));
+        }
+        ProjectEntity projectEntity = projectInvitationTokenEntity.getProject();
+        UserEntity userEntity = projectInvitationTokenEntity.getUser();
+        ProjectEntity projectWithSavedUser = projectSLO.addUserToProject(projectEntity, userEntity);
```

Comment:
To nie powinno byc za tym ifem? Bo teraz doda usera nawet jeśli token nie jest jego